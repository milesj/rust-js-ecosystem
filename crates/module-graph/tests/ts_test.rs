mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::*;

mod ts {
    use super::*;

    #[test]
    fn export_star() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "ts/export-star.ts"));
    }

    #[test]
    fn export_named() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-named.ts"
        ));
    }

    #[test]
    fn export_default_function() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-func.ts"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-anon-func.ts"
        ));
    }

    #[test]
    fn export_default_class() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-class.ts"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-anon-class.ts"
        ));
    }

    #[test]
    fn export_default_reference() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-ref.ts"
        ));
    }

    #[test]
    fn export_default_interface() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-interface.ts"
        ));
    }

    // This doesn't seem to generate exports,
    // even though oxc is a valid enum type!
    #[test]
    fn export_default_enum() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-enum.ts"
        ));
    }
}
