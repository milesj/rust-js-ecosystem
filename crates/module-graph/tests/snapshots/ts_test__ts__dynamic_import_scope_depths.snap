---
source: crates/module-graph/tests/ts_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"ts/dyn-import-scopes.ts\")"
---
{
    2: Module {
        exports: [
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 0,
                        end: 26,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "number",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 27,
                        end: 55,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "string",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 56,
                        end: 83,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: ValueType,
                        symbol_id: None,
                        name: "Type",
                    },
                ],
                type_only: true,
            },
        ],
        fragment: None,
        id: 2,
        imports: [],
        package_name: None,
        path: "/root/ts/shared.ts",
        query: None,
        source: JavaScriptModule {
            package_type: Unknown,
            source: "export const number = 123;\nexport const string = \"abc\";\nexport type Type = boolean;\n",
            source_type: SourceType {
                language: TypeScript {
                    is_definition_file: false,
                },
                module_kind: Module,
                variant: Standard,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 0,
                export_statements: 3,
                exports_default: false,
                import_statements: 0,
                other_statements: 0,
                require_count: 0,
            },
        },
    },
    1: Module {
        exports: [],
        fragment: None,
        id: 1,
        imports: [
            Import {
                kind: AsyncDynamic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 18,
                    end: 36,
                },
                symbols: [],
                type_only: false,
            },
            Import {
                kind: AsyncDynamic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 81,
                    end: 99,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Namespace,
                        source_name: None,
                        symbol_id: None,
                        name: "ns",
                    },
                ],
                type_only: false,
            },
            Import {
                kind: AsyncDynamic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 179,
                    end: 197,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Value,
                        source_name: None,
                        symbol_id: None,
                        name: "number",
                    },
                ],
                type_only: false,
            },
            Import {
                kind: AsyncDynamic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 267,
                    end: 285,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Value,
                        source_name: None,
                        symbol_id: None,
                        name: "string",
                    },
                ],
                type_only: false,
            },
        ],
        package_name: None,
        path: "/root/ts/dyn-import-scopes.ts",
        query: None,
        source: JavaScriptModule {
            package_type: Unknown,
            source: "export {};\n\nawait import(\"./shared\");\n\nasync function foo() {\n  const ns = await import(\"./shared\");\n}\n\nasync function bar() {\n  return async () => {\n    const { number } = await import(\"./shared\");\n  };\n}\n\nclass Baz {\n  async method() {\n    const { string } = await import(\"./shared\");\n  }\n}\n",
            source_type: SourceType {
                language: TypeScript {
                    is_definition_file: false,
                },
                module_kind: Module,
                variant: Standard,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 4,
                export_statements: 1,
                exports_default: false,
                import_statements: 0,
                other_statements: 3,
                require_count: 0,
            },
        },
    },
}
